languages/translations/hu.json

```diff
@@ -0,0 +1,493 @@
+{
+    "meta": {
+        "capitalizeFirstLetter": true
+    },
+    "v5": {
+        "constants": {
+            "ordinalize": {
+                "1": "1.",
+                "2": "2.",
+                "3": "3.",
+                "4": "4.",
+                "5": "5.",
+                "6": "6.",
+                "7": "7.",
+                "8": "8.",
+                "9": "9.",
+                "10": "10."
+            },
+            "direction": {
+                "north": "észak",
+                "northeast": "északkelet",
+                "east": "kelet",
+                "southeast": "délkelet",
+                "south": "dél",
+                "southwest": "délnyugat",
+                "west": "nyugat",
+                "northwest": "északnyugat"
+            },
+            "modifier": {
+                "left": "balra",
+                "right": "jobbra",
+                "sharp left": "élesen balra",
+                "sharp right": "élesen jobbra",
+                "slight left": "enyhén balra",
+                "slight right": "enyhén jobbra",
+                "straight": "egyenesen",
+                "uturn": "vissza"
+            },
+            "lanes": {
+                "xo": "Tartson jobbra",
+                "ox": "Tartson balra",
+                "xox": "Maradjon középen",
+                "oxo": "Tartson balra vagy jobbra"
+            }
+        },
+        "modes": {
+            "ferry": {
+                "default": "Hajtson fel a kompra",
+                "name": "Hajtson fel a kompra a(z) {way_name} felé",
```

Comment:
According to [Wiktionary](https://en.wiktionary.org/wiki/az#Article), it would be “a” or “az” depending on the first letter of `way_name`. `a(z)` likely won’t sound right when spoken by a text-to-speech engine. We should add a simple grammar rule that changes it to “a” or “az” based on `way_name`.

Reply:
Yes, and @kiskunk and I agreed to try to do this in separate PR. But I could try to include this into already prepared. I just worry that PR may become too huge to be reviewed completely.

Reply:
@kiskunk, would these regular expressions correctly match all the vowels and consonants that could appear at the beginning of a name?

```json
"article": [
    ["^ [aáäåâeéëiíïoóőöuúűüyÿ]", " az "],
    ["^ [^aáäåâeéëiíïoóőöuúűüyÿ]", " a "]
]
```

There would also need to be an override file similar to the French one:

https://github.com/Project-OSRM/osrm-text-instructions/blob/ed1a4a97647e6bb18eea4bcc8fa098c7d9f35f20/languages/overrides/fr.js#L3-L9

Reply:
> @kiskunk, would these regular expressions correctly match all the vowels and consonants that could appear at the beginning of a name?
> 
> ```json
> "article": [
>     ["^ [aáäåâeéëiíïoóőöuúűüyÿ]", " az "],
>     ["^ [^aáäåâeéëiíïoóőöuúűüyÿ]", " a "]
> ]
> ```
> 
> There would also need to be an override file similar to the French one:
> 
> [osrm-text-instructions/languages/overrides/fr.js](https://github.com/Project-OSRM/osrm-text-instructions/blob/ed1a4a97647e6bb18eea4bcc8fa098c7d9f35f20/languages/overrides/fr.js#L3-L9)
> 
> Lines 3 to 9 in [ed1a4a9](/Project-OSRM/osrm-text-instructions/commit/ed1a4a97647e6bb18eea4bcc8fa098c7d9f35f20)
> 
>  var replaces = [ 
>      [' (de )?\{destination\}', ' {destination:preposition}'], // eslint-disable-line no-useless-escape 
>      [' (le rond-point )?\{rotary_name\}', ' {rotary_name:rotary}'], // eslint-disable-line no-useless-escape 
>      [' fin (de )?(la route )?\{way_name\}', ' fin {way_name:preposition}'], // eslint-disable-line no-useless-escape 
>      [' \{way_name\}', ' {way_name:article}'], // eslint-disable-line no-useless-escape 
>      [' (à )?\{waypoint_name\}', ' {waypoint_name:arrival}'] // eslint-disable-line no-useless-escape 
>  ];

It would be good for most of the time, but not always, if there is any way to determine that a way_name/destination is a highway or not, it could be helpful, without that the globally accepted "a(z)" format is ok.

Reply:
So it seems we need add rules for Hungarian grammatical cases where we will recognize status street names and also change `a/az` articles appropriately. We can treat all unrecognized as highway refs and always use `az` there (but all foreign street names will be also here 😢) or add special rules to match "typical" highway refs (capital letter with digits or digits only, etc).

Currently I know 11 Hungarian status street names, see https://github.com/yuryleb/garmin-russian-tts-voices/blob/121a9bfe49eba702d5f244cd34ec8b0b7201aabb/src/Pycckuu__Milena%202.10/RP_RSET.TXT#L482-L493
Maybe you could extend this list more?

Also what grammatical cases could be applied for Hungarian messages? For example Russian localization supports 4 cases (accusative, dative, genitive and prepositional).

Reply:
Wait, so whether the instruction uses “a” or “az” depends on whether it’s a motorway? Or does it depend on some other aspect of the road related to motorways, like whether there’s a `ref` instead of a `name`?